src/pr-comment/pr-comment.ts

```diff
@@ -0,0 +1,192 @@
+import * as fs from "fs";
+import * as path from "path";
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import { getErrorMessage } from "../utils";
+
+/*********/
+/* TYPES */
+/*********/
+enum ResultType {
+    Patch = "patch",
+    Minor = "minor",
+    Major = "major",
+    Warning = "warning",
+    Error = "error",
+}
+
+class ResultSummary {
+    minors: CargoSemverChecksMinor[];
+    majors: CargoSemverChecksMajor[];
+    warnings: CargoSemverChecksWarning[];
+    errors: CargoSemverChecksError[];
+    result: ResultType;
+
+    constructor(
+        minors: CargoSemverChecksMinor[],
+        majors: CargoSemverChecksMajor[],
+        warnings: CargoSemverChecksWarning[],
+        errors: CargoSemverChecksError[],
+    ) {
+        this.minors = minors;
+        this.majors = majors;
+        this.warnings = warnings;
+        this.errors = errors;
+
+        this.result = this.determineResultType();
+    }
+
+    toString(): string {
+        return `ResultSummary:
+        Result: ${this.result}
+        Minors (${this.minors.length}): ${this.minors}
+        Majors (${this.majors.length}): ${this.majors}
+        Warnings (${this.warnings.length}): ${this.warnings}
+        Errors (${this.errors.length}): ${this.errors}`;
+    }
+
+    private determineResultType(): ResultType {
+        if (this.errors.length) {
+            return ResultType.Error;
+        } else if (this.warnings.length) {
+            return ResultType.Warning;
+        } else if (this.majors.length) {
+            return ResultType.Major;
+        } else if (this.minors.length) {
+            return ResultType.Minor;
+        } else {
+            return ResultType.Patch;
+        }
```

Comment:
I'm a bit confused about this hierarchy. Could you help me map this to `cargo-semver-checks` outcomes?

In c-s-c, lints can be semver-major or semver-minor, and when they trigger, they can be treated as errors or warnings. In principle, I guess it's possible that c-s-c completely fails to run for some reason, so that's an additional outcome to consider. How does that connect to the code here?

Reply:
I might have misinterpreted this myself. After adjusting for real outcomes this should become clear.